src/snowflake/cli/plugins/nativeapp/lsp_commands.py

```diff
@@ -0,0 +1,88 @@
+from typing import TypedDict
+
+from pygls.server import LanguageServer
+from snowflake.cli.api.output.types import MessageResult
+from snowflake.cli.api.project.definition_manager import DefinitionManager
+from snowflake.cli.plugins.nativeapp.init import nativeapp_init
+from snowflake.cli.plugins.nativeapp.manager import NativeAppManager
+from snowflake.cli.plugins.rpc.manager import LSPPluginContext
+from snowflake.snowpark.dataframe import DataFrame
+
+
+def lsp_plugin(name, version, capabilities):
+    def _decorator(func):
+        func.is_lsp_plugin = True
+        func.lsp_plugin_name = name
+        func.lsp_plugin_version = version
+        func.lsp_plugin_capabilities = capabilities
+        return func
+
+    return _decorator
+
+
+class ConnectionParams(TypedDict):
+    sessionToken: str
+    masterToken: str
+    account: str
+    host: str
+    port: str
+    protocol: str
+
+
+class CreateApplication(TypedDict):
+    workspacePath: str
+    applicationName: str
+    templateRepo: str
+    templateName: str
+
+
+class OpenApplication(TypedDict):
+    project_path: str
+
+
+@lsp_plugin(
+    name="nativeapp",
+    version="0.0.1",
+    capabilities={
+        "createApplication": True,
+        "openApplication": True,
+        "teardownApplication": True,
+    },
```

Comment:
Should this be an enum? We are duplicating the naming here and in `command` decorators